src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -18,7 +18,7 @@ public class FakeTimeProvider : TimeProvider
     internal readonly HashSet<Waiter> Waiters = new();
     private DateTimeOffset _now = new(2000, 1, 1, 0, 0, 0, 0, TimeSpan.Zero);
     private TimeZoneInfo _localTimeZone = TimeZoneInfo.Utc;
-    private int _wakeWaitersGate;
+    private volatile int _wakeWaitersGate;
```

Comment:
Didn't think volatile was needed when you read the value using Interlocked.CompareExchange?

Is it because it is being set by direct assignment later?

Reply:
> Didn't think volatile was needed when you read the value using Interlocked.CompareExchange?
> 
> Is it because it is being set by direct assignment later?

Yes, just as an added precaution against funny business in the compiler.